contentgrid-appserver-autoconfigure/src/main/java/com/contentgrid/appserver/autoconfigure/query/engine/JOOQQueryEngineAutoConfiguration.java

```diff
@@ -41,8 +46,20 @@ JOOQCountStrategy jooqTimedCountStrategy(@Value("${contentgrid.appserver.query-e
     }
 
     @Bean
-    QueryEngine jooqQueryEngine(DSLContextResolver dslContextResolver, JOOQCountStrategy countStrategy, PlatformTransactionManager transactionManager) {
-        return new TransactionalQueryEngine(new JOOQQueryEngine(dslContextResolver, countStrategy), transactionManager);
+    // When we run with the initContainer profile, we won't have real event handlers available
+    @ConditionalOnMissingBean(EventConsumer.class)
+    EventConsumer noopEventHandlers() {
+        return new EventConsumer() {
+            public void dispatchCreate(Application application, EntityName entity, EntityData data) {}
+            public void dispatchUpdate(Application application, EntityName entity, EntityData oldData, EntityData newData) {}
+            public void dispatchDelete(Application application, EntityName entity, EntityData oldData) {}
+        };
+    }
+
+    @Bean
+    QueryEngine jooqQueryEngine(DSLContextResolver dslContextResolver, JOOQCountStrategy countStrategy,
+            PlatformTransactionManager transactionManager, EventConsumer eventHandlers) {
+        return new TransactionalQueryEngine(new JOOQQueryEngine(dslContextResolver, countStrategy, eventHandlers), transactionManager);
```

Comment:
`eventHandlers` should not be injected into the infrastructure-layer.

Instead, the `DatamodelApiImpl` domain-level service should pass an event handling object to the `QueryEngine` as a parameter every time that it calls a method.

That way, the infrastructure does not have a direct dependency on the domain. Instead, a domain-layer service is provided to the infrastructure when it needs it.